srml/contract/COMPLEXITY.md

```diff
@@ -158,20 +176,37 @@ Assuming marshaled size of a balance value is of the constant size we can neglec
 
 This function receives input data for the contract execution. The execution consists of the following steps:
 
-1. Loading code from the DB.
-2. `transfer`-ing funds between the caller and the destination account.
-3. Executing the code of the destination account.
-4. Committing overlayed changed to the underlying `AccountDb`.
+1. Checking rent payment.
+2. Loading code from the DB.
+3. `transfer`-ing funds between the caller and the destination account.
+4. Executing the code of the destination account.
+5. Committing overlayed changed to the underlying `AccountDb`.
 
 **Note** that the complexity of executing the contract code should be considered separately.
 
-Loading code most probably will trigger a DB read, since the code is immutable and therefore will not get into the cache (unless a suicide removes it).
+Checking for rent involves an unconditional DB read of `ContractInfoOf`
+and at most once per block:
+
+- DB read to `block_number` and
+- `free_balance` and
+- `rent_deposit_offset` and
+- `rent_byte_price` and
+- `Currency::minimum_balance` and
+- `tobmstone_deposit`.
+- Calls to `ensure_can_withdraw`, `withdraw`, `make_free_balance_be` can perform arbitrary logic and should be considered separately,
+- `child_storage_root`
+- `kill_child_storage`
+- mutation of `ContractInfoOf`
+
+Loading code most likely will trigger a DB read, since the code is immutable and therefore will not get into the cache (unless a suicide removes it).
```

Comment:
```suggestion
Loading code most likely will trigger a DB read, since the code is immutable and therefore will not get into the cache (unless a suicide removes it, or it has been created in the same call chain).
```

Reply:
Good point!